### PR TITLE
normalize file paths to get correct analysis results

### PR DESCRIPTION
**Pull request by Trass3r**
See #8.

### CODE REVIEW

File: src/Utils.cpp [hunk 1]
```diff
@@ -103,13 +106,7 @@ bool utils::IsHeader(const std::string_view& path)
...
 std::string utils::GetNicePath(const std::string_view& path)
 {
-    std::string res(path);
-    ForwardSlashify(res);
-    if (BeginsWith(res, s_CurrentDir))
-        res = res.substr(s_CurrentDir.size(), res.size() - s_CurrentDir.size());
-    if (BeginsWith(res, s_Root))
-        res = res.substr(s_Root.size(), res.size() - s_Root.size());
-    return res;
+    return fs::path(path).lexically_normal();
```
Comment (Trass3r) [inline src/Utils.cpp]: This code might produce incorrect paths if crazy symlinks are present.
But getting this right would require lots of file IO.

File: src/Utils.cpp [hunk 2]
```diff
@@ -103,13 +106,7 @@ bool utils::IsHeader(const std::string_view& path)
 
 std::string utils::GetNicePath(const std::string_view& path)
 {
-    std::string res(path);
-    ForwardSlashify(res);
-    if (BeginsWith(res, s_CurrentDir))
-        res = res.substr(s_CurrentDir.size(), res.size() - s_CurrentDir.size());
```
Comment (Trass3r) [inline src/Utils.cpp:109]: For me the workspace paths were already relative.
But I doubt this is useful in general. The cwd is most probably either the build dir or something else entirely. The project folder is the one that should be used, if any.

**Comment by Trass3r** [comment]
Looks like this got auto-closed when renaming the main branch.

**Comment by aras-p** [comment]
Oh sorry, did not realize that would happen! Will try to take a look next week.

**Comment by Trass3r** [comment]
And updates?

**Pull request closed without merging**